### DISCUSSION TITLE
Query on Utilizing kwargs in Vector Store query Methods

### DISCUSSION THREAD

**Discussion by Izukimat** [run-llama/llama_index#10448] [Ideas]
**Context and Code Example:**
I have been working with the MilvusVectorStore, and I observed a pattern that seems prevalent in other vector store implementations as well. The query method in these classes doesn't utilize additional arguments passed via kwargs to influence query parameters like output_fields. Following is a code snippet from the MilvusVectorStore for illustration:
```
def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:
        """Query index for top k most similar nodes.

        Args:
            query_embedding (List[float]): query embedding
            similarity_top_k (int): top k most similar nodes
            doc_ids (Optional[List[str]]): list of doc_ids to filter by
            node_ids (Optional[List[str]]): list of node_ids to filter by
            output_fields (Optional[List[str]]): list of fields to return
            embedding_field (Optional[str]): name of embedding field
        """
    # ... [some code] ...

    # The method uses properties from the query object directly
    # but does not incorporate kwargs into these properties
    output_fields = ["*"]
    if query.output_fields is not None:
        output_fields = query.output_fields

    # ... [rest of the method] ...

```
In this implementation, `output_fields` and similar parameters are sourced directly from the query object. This means that even if I pass output_fields dynamically through kwargs, it has no effect on query.output_fields or the behaviour of the method. 

**Question and Possible Contribution:**
**Dynamic Parameter Handling:** Is there a preferred approach or existing mechanism to dynamically set properties like output_fields for a query, especially when using kwargs?
**Initial Setting of query.output_fields:** How is query.output_fields initially set or determined within the query method? 

If this is an area open for enhancement, I am interested in contributing by extending the query methods to better handle dynamic parameters via kwargs. Are there any considerations I should be aware of in this process?

**Comment by nerdai** [comment]
Thanks @Izukimat. Yea improved kwargs propagation would be great. Have any ideas on you might do this?
